docs/pipelines/overview/configuration.md

```diff
@@ -1,95 +1,368 @@
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+In this example, all the units located within the `an-account` directory sibling to the `.gruntwork` directory will assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands by Pipelines.
+
+A typical approach to building Pipelines configurations is to first define minimal configurations that address the most common use-cases, and then to refactor and generalize those configurations as needed to reduce repetition.
+
+More details on how these configurations are defined will be detailed below.
+
+## Global Configurations
+
+The configurations found within a `.gruntwork` directory found in the current working directory, or a parent directory of the current working directory are referred to as global configurations. These configurations are typically applicable within a wide range of contexts within a repository, and are the primary mechanism for configuring Pipelines.
+
+Pipelines will attempt to find exactly one directory named `.gruntwork` when it is attempting to discover configurations. It will not continue to search for configurations in parent directories once it finds a `.gruntwork` directory.
+
+Note that you will frequently see filenames for configurations within the `.gruntwork` directory that are named after the configuration block that they define. This is a common pattern, but not a requirement. Any `.hcl` file found within a `.gruntwork` directory will be parsed by Pipelines as one global configuration.
+
+### Environment Blocks
+
+Environment blocks are used to define configurations that are applicable to a specific environment within a repository.
+
+The label applied to an environment block is the name of the environment. This is a user-defined label for the environment, and must be globally unique.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_environment" {
+  filter {
+    paths = ["an-environment/*"]
+  }
+
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role  = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+In this example, the `an_environment` environment is defined to match all units located within the `an-environment` directory sibling to the `.gruntwork` directory. All units that match this filter will assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands by Pipelines.
+
+The environment block is most idiomatically used in conjunction with other configuration blocks, as the contents frequently involve configurations that are applicable to multiple units within a repository.
```

Comment:
The wording of this sentence seems unnecessarily complex.  "Environments should reference other blocks rather than redefine configuration everytime."